superset-frontend/src/components/DeprecatedSelect/DeprecatedSelect.tsx

```diff
@@ -161,13 +161,15 @@ function styled<
       filterOption,
       ignoreAccents = false, // default is `true`, but it is slow
 
+      asText = (value: any) => (value ? String(value) : ''),
```

Comment:
when value is `0`, it will returns `''`.  Please use nullish coalescing operator(`??`) to avoid the case.
```suggestion
      asText = (value: any) => String(value ?? ''),
```